module_utils/yc.py

```diff
@@ -27,16 +27,20 @@ def yc_argument_spec():
 
 class YC(AnsibleModule):
     def __init__(self, *args, **kwargs):
-
         argument_spec = yc_argument_spec()
         argument_spec.update(kwargs.get("argument_spec", dict()))
         kwargs["argument_spec"] = argument_spec
-
         super().__init__(*args, **kwargs)
         interceptor = RetryInterceptor(max_retry_count=10)
         if self.params["auth"]["root_certificates"]:
             self.params["auth"]["root_certificates"] = self.params["auth"]["root_certificates"].encode("utf-8")
         self.sdk = SDK(interceptor=interceptor, **self.params["auth"])
+        if self.params.get("fqdn"):
+            self.params["name"] = self.params["fqdn"].split('.')[0]
+            self.params["hostname"] = self.params["fqdn"].split('.')[0]
+        else:
+            self.params["name"] = self.params["name"].split('.')[0]
+            self.params["hostname"] = self.params["name"].split('.')[0]
 
```

Comment:
Do we need to split self.params["name"] in else branch?  I think it's better to validate name and hostname and raise an exception if an incorrect value is passed. Please look at validation of hostname add_vm method, I think we can move that validation check into constructor.

Reply:
@mooncube I believe split is necessary because in our bundle we can't specify the name and fqdn will be passed to ansible module as name. So we have to do split at some point. 
I'll move validation of hostname to constructor